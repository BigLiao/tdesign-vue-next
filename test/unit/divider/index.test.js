import { mount } from '@vue/test-utils';
import Divider from '@/src/divider/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Divider', () => {
  // test props api
  describe(':props', () => {
    it(':type', () => {
      const wrapper = mount({
        render() {
          return <Divider type="vertical"></Divider>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':dashed', () => {
      const wrapper = mount({
        render() {
          return <Divider dashed></Divider>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':orientation', () => {
      const wrapper = mount({
        render() {
          return <Divider orientation="left">TDesign</Divider>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  // test slots
  describe('<slot>', () => {
    it('default', () => {
      const wrapper = mount({
        render() {
          return <Divider>TDesign</Divider>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
