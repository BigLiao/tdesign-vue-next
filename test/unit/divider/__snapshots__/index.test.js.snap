// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider :props :dashed 1`] = `<div class="t-divider t-divider--horizontal t-divider--dashed"></div>`;

exports[`Divider :props :orientation 1`] = `<div class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-left"><span classname="t-divider-inner-text">腾讯左</span></div>`;

exports[`Divider :props :type 1`] = `<div class="t-divider t-divider--vertical"></div>`;

exports[`Divider <slot> default 1`] = `<div class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"><span classname="t-divider-inner-text">腾讯中</span></div>`;
